easybuild/tools/options.py

```diff
@@ -383,9 +384,10 @@ def override_options(self):
                              str, 'store', False, {'metavar': "PATH"}),
             'use-existing-modules': ("Use existing modules when resolving dependencies with minimal toolchains",
                                      None, 'store_true', False),
+            'verify-easyconfig-filenames': ("Verify whether filename of specified easyconfigs matches with contents",
```

Comment:
No, that would change the behaviour, sites should enable this if they want to stricter behaviour.
I don't consider the flexibility of filenames for specified easyconfigs a bug, but a feature. I'd get annoyed fi EasyBuild gets picky about filenames when it doesn't really matter; it only really matters for the dependency resolver, which already complains if an easyconfig file that it picks up by filename doesn't contain what is expected.

We can reconsider this in EasyBuild v4.0, but I'm in favour of leaving the default behaviour as is.